helm-charts/images/hub/Dockerfile

```diff
@@ -18,9 +18,8 @@ RUN pip install --no-cache git+https://github.com/yuvipanda/jupyterhub-configura
 # z2jh 1.2.x ships with kubespawner 1.1.0, so we just do a little bump
 RUN pip install --no-cache --upgrade jupyterhub-kubespawner==1.1.2
 
-# Brings in https://github.com/jupyterhub/oauthenticator/pull/498 and
-# https://github.com/jupyterhub/oauthenticator/pull/504
-RUN pip install --no-cache --upgrade git+https://github.com/jupyterhub/oauthenticator@60ddc4fe3ab06b19c6876992679c1fd2ec1c64dd
+# Latest version comes with some breaking changes https://oauthenticator.readthedocs.io/en/latest/migrations.html#migrating-cilogonoauthenticator-to-version-15-0
+RUN pip install --no-cache --upgrade oauthenticator==15.0.1
```

Comment:
What is the effect of having the `--upgrade` flag but a strictly pinned package here? I feel like either the flag has no effect and maybe we should remove it, or it could have strange effects? Hmmm....

Reply:
Hmm, I think you're right and this flag might be not be needed in this case of pinned pkgs. But I don't believe it does any harm, but rather does nothing at the moment.

I think we can remove it. The only reason I can think of that would be in favor of keeping it is if we ever decide to not use a pinned version of these pkgs and we forget to add the flag, but both things seem unlikely.

Do you think I should remove it as part of this PR? (it's also used when installing `kubespawner` so that would need to be removed too) or in its own thing?

Reply:
>The only reason I can think of that would be in favor of keeping it is if we ever decide to not use a pinned version of these pkgs and we forget to add the flag

I am +1 on this.

I am fine keeping the flag if we are confident it does no harm. But just wanted to check :)